src/util/UserphraseSet.h

```diff
@@ -50,8 +50,17 @@ class UserphraseSet {
     const Userphrase& operator[](size_t index) const;
 
     void swap(UserphraseSet& set) noexcept;
+    
+    // shadow relative functionality
+    void insert_shadow(Userphrase&& userphrase);
+    void erase_shadow(iterator it);
+    
+    iterator end_shadow();
+
+    bool empty_shadow() const { return userphrase_shadow_.empty(); }
 
 private:
     std::vector<Userphrase> userphrase_;
+    std::vector<Userphrase> userphrase_shadow_;
     std::set<QString> dedup_;
 };
```

Comment:
@jserv : I am not sure is this a good way. Please give me some advice, thank you!